source/fundamentals/builders/aggregation-expression-operations.txt

```diff
@@ -0,0 +1,1198 @@
+     - *No corresponding operator*
+
+   * - `getDocument() <{+api+}/apidocs/mongodb-driver-core/com/mongodb/client/model/mql/MqlDocument.html#getDocument(java.lang.String)>`__
+     - *No corresponding operator*
+
+   * - `getField() <{+api+}/apidocs/mongodb-driver-core/com/mongodb/client/model/mql/MqlDocument.html#getField(java.lang.String)>`__
+     - :manual:`$getField </reference/operator/aggregation/getField/>`
+
+   * - `getInteger() <{+api+}/apidocs/mongodb-driver-core/com/mongodb/client/model/mql/MqlDocument.html#getInteger(java.lang.String)>`__
+     - *No corresponding operator*
+
+   * - `getMap() <{+api+}/apidocs/mongodb-driver-core/com/mongodb/client/model/mql/MqlDocument.html#getMap(java.lang.String)>`__
+     - *No corresponding operator*
+
+   * - `getNumber() <{+api+}/apidocs/mongodb-driver-core/com/mongodb/client/model/mql/MqlDocument.html#getNumber(java.lang.String)>`__
+     - *No corresponding operator*
+
+   * - `getString() <{+api+}/apidocs/mongodb-driver-core/com/mongodb/client/model/mql/MqlDocument.html#getString(java.lang.String)>`__
+     - *No corresponding operator*
+
+   * - `hasField() <{+api+}/apidocs/mongodb-driver-core/com/mongodb/client/model/mql/MqlDocument.html#hasField(java.lang.String)>`__
+     - *No corresponding operator*
+
+   * - `merge() <{+api+}/apidocs/mongodb-driver-core/com/mongodb/client/model/mql/MqlDocument.html#merge(com.mongodb.client.model.mql.MqlDocument)>`__
+     - :manual:`$mergeObjects </reference/operator/aggregation/mergeObjects/>`
+
+   * - `setField() <{+api+}/apidocs/mongodb-driver-core/com/mongodb/client/model/mql/MqlDocument.html#setField(java.lang.String,com.mongodb.client.model.mql.MqlValue)>`__
+     - :manual:`$setField </reference/operator/aggregation/setField/>`
+
+   * - `unsetField() <{+api+}/apidocs/mongodb-driver-core/com/mongodb/client/model/mql/>`__
+     - :manual:`$unsetField </reference/operator/aggregation/unsetField/MqlDocument.html#unsetField(java.lang.String)>`
+
+Suppose you have a collection of customer data which includes their current and
+addresses as a nested document. You want to find all the customers who
+currently live in Washington state. A document in this collection might
+resemble the following:
+
+.. code-block:: json
+   :copyable: false
+
+   {
+     "_id": ...,
+     "customerName": "Elizabeth Ray",
+     "currentAddress":
+     {
+       "street": "9286 Bethany Glens",
+       "city": "Vasqueztown",
+       "state": "CO"
+       "zip": 22939
+     }
+   }
+
+The ``getDocument()`` method retrieves the nested ``currentAddress`` document
+so the state can be retrieved with the ``getString()`` method. The ``eq()``
+method compares the string to ``"WA"`` to see if the state matches.
+
+The following code shows the pipeline for this aggregation:
+
+.. code-block:: java
+   :copyable: true
+
+   asList(match(
+         expr(current()
+               .getDocument("currentAddress")
+               .getString("state")
```

Comment:
I wonder if we should include a field that has a `.` in it to demonstrate that these `get...` methods handle those properly and do not support "paths" in the way that raw MQL JSON does (like `$currentAddress.state` below, which is not a single field with a dot but one path with two fields).

Reply:
Agreed, but the deviation would be intentional to demonstrate a "gotcha", and we can mention it in the wording "...has legacy data that uses a `.` in the field name..."